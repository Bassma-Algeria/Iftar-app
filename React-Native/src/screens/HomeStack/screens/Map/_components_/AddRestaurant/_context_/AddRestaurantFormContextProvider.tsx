import React, {createContext, useState} from 'react';
import type {RestaurantInfoToAdd} from '../../../../../../../Gateways/RestaurantsGateway/RestaurantsGateway.interface';

interface AddRestaurantContextValues {
  restaurantInfo: RestaurantInfoToAdd;
  setRestaurantInfo: React.Dispatch<React.SetStateAction<RestaurantInfoToAdd>>;

  serverError: boolean;
  setServerError: React.Dispatch<React.SetStateAction<boolean>>;
}

const AddRestaurantFormContext = createContext<AddRestaurantContextValues | undefined>(undefined);

const AddRestaurantFormContextProvider: React.FC = ({children}) => {
  const [serverError, setServerError] = useState<boolean>(false);
  const [restaurantInfo, setRestaurantInfo] = useState<RestaurantInfoToAdd>({
    name: '',
    ownerName: '',
    pictures: [],
    closingTime: '',
    openingTime: '',
    locationCoords: undefined,
    placeName: '',
  });

  return (
    <AddRestaurantFormContext.Provider
      value={{
        restaurantInfo,
        setRestaurantInfo,
        serverError,
        setServerError,
      }}>
      {children}
    </AddRestaurantFormContext.Provider>
  );
};

export {AddRestaurantFormContextProvider, AddRestaurantFormContext};
