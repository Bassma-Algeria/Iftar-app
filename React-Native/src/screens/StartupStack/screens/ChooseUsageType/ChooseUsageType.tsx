import React from 'react';
import {View} from 'react-native';

import {styles} from './ChooseUsageType.style';

import {Layout} from '../../_components_/Layout/Layout';
import {Logo} from '../../../../components/Logo/Logo';
import {Button} from '../../../../components/Button/Button';

import type {MainStackScreenProps} from '../../../MainStack.types';

interface Props extends MainStackScreenProps<'StartupStack'> {}

const ChooseUsageType: React.FC<Props> = ({navigation}) => {
  return (
    <>
      <Layout>
        <View style={styles.container}>
          <Logo />
        </View>
      </Layout>

      <View style={styles.buttonsContainer}>
        <Button onPress={() => navigation.navigate('HomeStack', {screen: 'Map'})}>
          إبحث عن مطعم
        </Button>

        <View style={styles.buttonsSeparator} />

        <Button onPress={() => navigation.navigate('StartupStack', {screen: 'Login'})}>
          أضف مطعم
        </Button>
      </View>
    </>
  );
};

export {ChooseUsageType};
