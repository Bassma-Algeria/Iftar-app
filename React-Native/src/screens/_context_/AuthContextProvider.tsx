import React, {createContext, useState} from 'react';

interface AuthContextValues {
  isRestaurantOwner: boolean;
  setIsRestaurantOwner: React.Dispatch<React.SetStateAction<boolean>>;
}

const AuthContext = createContext<AuthContextValues | undefined>(undefined);

const AuthContextProvider: React.FC = ({children}) => {
  const [isRestaurantOwner, setIsRestaurantOwner] = useState<boolean>(false);

  return (
    <AuthContext.Provider value={{isRestaurantOwner, setIsRestaurantOwner}}>
      {children}
    </AuthContext.Provider>
  );
};

export {AuthContextProvider, AuthContext};
