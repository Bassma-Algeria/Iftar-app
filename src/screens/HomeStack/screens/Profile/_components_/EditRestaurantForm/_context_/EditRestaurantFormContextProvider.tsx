import React, {createContext, useLayoutEffect, useState} from 'react';

import {RestaurantInfo} from '../../../../../../../Gateways/RestaurantsGateway/RestaurantsGateway.interface';

interface IEditRestaurantForm extends Omit<RestaurantInfo, 'pictures'> {
  pictures: ({uri: string; base64: string} | string)[];
}

interface EditRestaurantContextValues {
  restaurantInfo: IEditRestaurantForm;
  setRestaurantInfo: React.Dispatch<React.SetStateAction<IEditRestaurantForm>>;
}

const EditRestaurantFormContext = createContext<EditRestaurantContextValues | undefined>(undefined);

const initialEditRestaurantFormState: IEditRestaurantForm = {
  ownerId: '',
  restaurantId: '',
  name: '',
  locationName: '',
  locationCoords: {latitude: 0, longitude: 0},
  workingTime: {opening: {hour: 0, minute: 0}, closing: {hour: 0, minute: 0}},
  pictures: [],
};

const EditRestaurantFormContextProvider: React.FC<{restaurant?: RestaurantInfo}> = ({
  children,
  restaurant,
}) => {
  const [restaurantInfo, setRestaurantInfo] = useState<IEditRestaurantForm>(
    initialEditRestaurantFormState,
  );

  useLayoutEffect(() => {
    if (restaurant) {
      setRestaurantInfo(restaurant);
    }
  }, [restaurant]);

  return (
    <EditRestaurantFormContext.Provider
      value={{
        restaurantInfo,
        setRestaurantInfo,
      }}>
      {children}
    </EditRestaurantFormContext.Provider>
  );
};

export {
  EditRestaurantFormContextProvider,
  EditRestaurantFormContext,
  initialEditRestaurantFormState,
};
