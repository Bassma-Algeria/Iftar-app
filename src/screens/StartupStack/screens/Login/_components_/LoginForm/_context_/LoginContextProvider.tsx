import React, {createContext, useState} from 'react';
import {LoginInfo} from '../../../../../../../Gateways/AuthGateway/AuthGateway.interface';

interface LoginContextValues {
  loginInfo: LoginInfo;
  setLoginInfo: React.Dispatch<React.SetStateAction<LoginInfo>>;

  serverError: string;
  setServerError: React.Dispatch<React.SetStateAction<string>>;
}

export const LoginContext = createContext<LoginContextValues | undefined>(undefined);

const LoginContextProvider: React.FC = ({children}) => {
  const [loginInfo, setLoginInfo] = useState<LoginInfo>({email: '', password: ''});
  const [serverError, setServerError] = useState<string>('');

  return (
    <LoginContext.Provider
      value={{
        loginInfo,
        setLoginInfo,
        serverError,
        setServerError,
      }}>
      {children}
    </LoginContext.Provider>
  );
};

export {LoginContextProvider};
