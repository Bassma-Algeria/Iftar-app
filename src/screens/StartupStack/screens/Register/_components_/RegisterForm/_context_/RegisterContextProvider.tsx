import React, {createContext, useState} from 'react';
import {SignupInfo} from '../../../../../../../Gateways/AuthGateway/AuthGateway.interface';

interface RegisterContextValues {
  registerInfo: SignupInfo;
  setRegisterInfo: React.Dispatch<React.SetStateAction<SignupInfo>>;

  serverError: string;
  setServerError: React.Dispatch<React.SetStateAction<string>>;
}

export const RegisterContext = createContext<RegisterContextValues | undefined>(undefined);

const RegisterContextProvider: React.FC = ({children}) => {
  const [registerInfo, setRegisterInfo] = useState<SignupInfo>({
    email: '',
    password: '',
    confirmPassword: '',
    phoneNumber: '',
  });
  const [serverError, setServerError] = useState<string>('');

  return (
    <RegisterContext.Provider
      value={{
        registerInfo,
        setRegisterInfo,
        serverError,
        setServerError,
      }}>
      {children}
    </RegisterContext.Provider>
  );
};

export {RegisterContextProvider};
