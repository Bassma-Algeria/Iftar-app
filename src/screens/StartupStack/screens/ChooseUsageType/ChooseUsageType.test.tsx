import React from 'react';
import renderer from 'react-test-renderer';
import {AuthContextProvider} from '../../../_context_/AuthContextProvider';

import {ChooseUsageType} from './ChooseUsageType';

describe('ChooseUsageType Screen ', () => {
  it('should render correctly', () => {
    const props: any = {};
    const tree = renderer
      .create(
        <AuthContextProvider>
          <ChooseUsageType {...props} />
        </AuthContextProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
