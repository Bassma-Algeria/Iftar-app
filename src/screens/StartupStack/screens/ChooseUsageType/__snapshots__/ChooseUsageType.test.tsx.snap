// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseUsageType Screen  should render correctly 1`] = `
Array [
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#F4E6CA",
          "height": "100%",
          "paddingTop": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "position": "absolute",
            "top": -30,
            "width": "100%",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "maxWidth": 200,
            }
          }
        />
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "maxWidth": 200,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginTop": "60%",
            "padding": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 30,
              "width": "100%",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "baseline",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#7C4908",
                      "fontFamily": "Cairo-Bold",
                      "fontSize": 35,
                    },
                    undefined,
                  ]
                }
              >
                إفطار
              </Text>
              <Image
                collapsable={false}
                nativeID="animatedComponent"
                resizeMode="contain"
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "left": 4,
                    "position": "absolute",
                    "top": -3,
                    "transform": Array [
                      Object {
                        "rotate": "0deg",
                      },
                    ],
                    "width": 20,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      Object {
        "bottom": 0,
        "padding": 15,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {},
        ]
      }
    >
      <View
        pointerEvents="box-none"
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth={3}
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": 30,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": 3,
                },
              ]
            }
            width={3}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="left"
                  x1="1"
                  x2="0"
                  y1="0"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "left",
                  ]
                }
                height="100%"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={2}
                x={0}
                y={-60}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth={3}
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": 30,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": 3,
                },
              ]
            }
            width={3}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="right"
                  x1="0"
                  x2="1"
                  y1="0"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "right",
                  ]
                }
                height="100%"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={2}
                x={0}
                y={-60}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={3}
            bbWidth="100%"
            focusable={false}
            height={3}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": 30,
                  "position": "absolute",
                  "top": "100%",
                },
                Object {
                  "flex": 0,
                  "height": 3,
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="bottom"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="1"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "bottom",
                  ]
                }
                height={2}
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="100%"
                x={-60}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={3}
            bbWidth="100%"
            focusable={false}
            height={3}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": 30,
                  "position": "absolute",
                  "top": -2,
                },
                Object {
                  "flex": 0,
                  "height": 3,
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="top"
                  x1="0"
                  x2="0"
                  y1="1"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "top",
                  ]
                }
                height={2}
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="100%"
                x={-60}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": -2,
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={32}
                  cy={32}
                  fx={32}
                  fy={32}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="topLeft"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M0,32 a32,32 0 0 1 32 -32 v2 a30,30 0 0 0 -30,30 h-2 Z"
                fill={
                  Array [
                    1,
                    "topLeft",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": -2,
                  "transform": Array [
                    Object {
                      "translateX": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={0}
                  cy={32}
                  fx={0}
                  fy={32}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="topRight"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M0,0 a32,32 0 0 1 32,32 h-2 a30,30 0 0 0 -30,-30 v-2 Z"
                fill={
                  Array [
                    1,
                    "topRight",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": "100%",
                  "transform": Array [
                    Object {
                      "translateY": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={32}
                  cy={0}
                  fx={32}
                  fy={0}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="bottomLeft"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M32,32 a32,32 0 0 1 -32,-32 h2 a30,30 0 0 0 30,30 v2 Z"
                fill={
                  Array [
                    1,
                    "bottomLeft",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": "100%",
                  "transform": Array [
                    Object {
                      "translateX": -30,
                    },
                    Object {
                      "translateY": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={0}
                  cy={0}
                  fx={0}
                  fy={0}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="bottomRight"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M32,0 a32,32 0 0 1 -32,32 v-2 a30,30 0 0 0 30,-30 h2 Z"
                fill={
                  Array [
                    1,
                    "bottomRight",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
              },
              undefined,
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#B87015",
                "borderRadius": 30,
                "flexDirection": "row-reverse",
                "height": 60,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 13,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFCF1",
                    "fontFamily": "Cairo-SemiBold",
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              إبحث عن مطعم
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 30,
        }
      }
    />
    <View
      style={
        Array [
          undefined,
          Object {},
        ]
      }
    >
      <View
        pointerEvents="box-none"
      >
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              undefined,
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth={3}
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": 30,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": 3,
                },
              ]
            }
            width={3}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="left"
                  x1="1"
                  x2="0"
                  y1="0"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "left",
                  ]
                }
                height="100%"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={2}
                x={0}
                y={-60}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth={3}
            focusable={false}
            height="100%"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": 30,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": 3,
                },
              ]
            }
            width={3}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="right"
                  x1="0"
                  x2="1"
                  y1="0"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "right",
                  ]
                }
                height="100%"
                propList={
                  Array [
                    "fill",
                  ]
                }
                width={2}
                x={0}
                y={-60}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={3}
            bbWidth="100%"
            focusable={false}
            height={3}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": 30,
                  "position": "absolute",
                  "top": "100%",
                },
                Object {
                  "flex": 0,
                  "height": 3,
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="bottom"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="1"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "bottom",
                  ]
                }
                height={2}
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="100%"
                x={-60}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={3}
            bbWidth="100%"
            focusable={false}
            height={3}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": 30,
                  "position": "absolute",
                  "top": -2,
                },
                Object {
                  "flex": 0,
                  "height": 3,
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    Array [
                      0,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="top"
                  x1="0"
                  x2="0"
                  y1="1"
                  y2="0"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  Array [
                    1,
                    "top",
                  ]
                }
                height={2}
                propList={
                  Array [
                    "fill",
                  ]
                }
                width="100%"
                x={-60}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": -2,
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup>
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={32}
                  cy={32}
                  fx={32}
                  fy={32}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="topLeft"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M0,32 a32,32 0 0 1 32 -32 v2 a30,30 0 0 0 -30,30 h-2 Z"
                fill={
                  Array [
                    1,
                    "topLeft",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": -2,
                  "transform": Array [
                    Object {
                      "translateX": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={0}
                  cy={32}
                  fx={0}
                  fy={32}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="topRight"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M0,0 a32,32 0 0 1 32,32 h-2 a30,30 0 0 0 -30,-30 v-2 Z"
                fill={
                  Array [
                    1,
                    "topRight",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": -2,
                  "position": "absolute",
                  "top": "100%",
                  "transform": Array [
                    Object {
                      "translateY": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={32}
                  cy={0}
                  fx={32}
                  fy={0}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="bottomLeft"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M32,32 a32,32 0 0 1 -32,-32 h2 a30,30 0 0 0 30,30 v2 Z"
                fill={
                  Array [
                    1,
                    "bottomLeft",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <RNSVGSvgView
            bbHeight={33}
            bbWidth={33}
            focusable={false}
            height={33}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "left": "100%",
                  "position": "absolute",
                  "top": "100%",
                  "transform": Array [
                    Object {
                      "translateX": -30,
                    },
                    Object {
                      "translateY": -30,
                    },
                  ],
                },
                Object {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            width={33}
          >
            <RNSVGGroup
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
            >
              <RNSVGDefs>
                <RNSVGRadialGradient
                  cx={0}
                  cy={0}
                  fx={0}
                  fy={0}
                  gradient={
                    Array [
                      0.9375,
                      -1632211950,
                      1,
                      11955218,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={1}
                  name="bottomRight"
                  rx={32}
                  ry={32}
                />
              </RNSVGDefs>
              <RNSVGPath
                d="M32,0 a32,32 0 0 1 -32,32 v-2 a30,30 0 0 0 30,-30 h2 Z"
                fill={
                  Array [
                    1,
                    "bottomRight",
                  ]
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
              },
              undefined,
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#B87015",
                "borderRadius": 30,
                "flexDirection": "row-reverse",
                "height": 60,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 13,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFCF1",
                    "fontFamily": "Cairo-SemiBold",
                    "fontSize": 20,
                  },
                  undefined,
                ]
              }
            >
              أضف مطعم
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
